__tests__/app.test.js

```diff
@@ -452,34 +469,88 @@ describe("GET /api/users", () => {
 				});
 		});
 	});
-	describe("error handling", () => {
-		test("responds with a 404 status when given a valid but non-existent endpoint", () => {
-			return request(app).get("/api/usersss").expect(404);
-		});
-	});
 });
 
+//// QUESTION 11 GetArticles by topic
 
 describe("GET /api/articles (topic query)", () => {
-	test("GET: should return articles filtered by topic if given a topic query", () => {
-		// const topic = "mitch";
-		return request(app)
-			.get("/api/articles?topic=mitch")
-			.expect(200)
-			.then((response) => {
-                const articles = response.body.articles;
-                console.log(articles)
-				expect(Array.isArray(articles)).toBe(true);
-				expect(articles).toBeSortedBy("created_at", { descending: true });
-				expect(articles.length).toBe(12);
-				articles.forEach((article) => {
-					expect(article.topic).toEqual("mitch");
+    describe("behaviours", () => {
+        test("GET: should return articles filtered by topic if given a topic query", () => {
+            return request(app)
+                .get("/api/articles?topic=mitch")
+                .expect(200)
+                .then((response) => {
+                    const articles = response.body.articles;
+                  
+                    expect(Array.isArray(articles)).toBe(true);
+                    expect(articles).toBeSortedBy("created_at", { descending: true });
+                    expect(articles.length).toBe(12);
+                    articles.forEach((article) => {
+                        expect(article.topic).toEqual("mitch");
+                    });
+                });
+        });
+    });
+
+    describe("error handling", () => {
+        test("responds with a 404 status when given a valid but non-existent query", () => {
+            return request(app).get("/api/articles?topic=witch").expect(404);
+        });
+    });
+})
+
+//// QUESTION 12 GET comment_count by article_id
+
+describe("GET /api/articles/article_id (comment_count)", () => {
+	describe("behaviours", () => {
+		test("GET: should return articles with comment count if accessing through a single comment via its id", () => {
+			return request(app)
+				.get("/api/articles/9")
+				.expect(200)
+				.then((response) => {
+                    const article = response.body.article;
+                    
+					expect(article.article_id).toBe(9);
+					expect(article.comment_count).toBe(2);
+
```

Comment:
This test is perfect up to this point. Everything below it is not needed as has been tested before 🙂